core/mutator/entry/mutation.go

```diff
@@ -148,3 +153,19 @@ func (m *Mutation) sign(signers []signatures.Signer) (*pb.Entry, error) {
 	m.entry.Signatures = sigs
 	return m.entry, nil
 }
+
+// Check verifies that an update was successfully applied.
+// Returns nil if newLeaf is equal to the entry in this mutation.
```

Comment:
It seems to me that the function is answering a yes/no question, but that errors can occur.
With this func signature the caller can't distinguish between 'no' and 'oops'.

A (bool, error) return would make it unambiguous but is ugly.

Better would be an error constant that the caller can check for.
You could just move the errors.New("message") to an constant in this package and return that.

Does Check need to be exported? I see it being used in this same file.  If so then the error constant should be too.

Reply:
`(bool, error)` is what I ended up doing.   
I thought about defining a public error, but then considered that it would force calling sites into a just as awkward error parsing. 

```
if err := m.Check(); err == ErrNotEqual {
} else if err != nil {
return err
}
```

Reply:
OK, I don't mind much either way, good you considered both options.